src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,179 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyDateException;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.EmptyIndexException;
+import duke.exception.IllegalDateFormatException;
+import duke.exception.IllegalInputException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidIndexException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * A parser class that encapsulates the action of parsing string to desired Object.
+ */
+public class Parser {
+
+    private static final int TASK_TYPE = 0;
+    private static final int COMMAND_DESCRIPTION = 1;
+
+    /***
+     * Parses the user input to relevant command.
+     * @param fullCommand User input
+     * @return Command that needs to be executed.
+     * @throws DukeException throws an exception when there is unexpected input.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] arr = fullCommand.trim().split("\\s+", 2);
+        String taskType = arr[TASK_TYPE];
+        switch (taskType.toUpperCase()) {
+        case "BYE":
+            return new ExitCommand();
```

Comment:
It's fantastic that your switch & case statements adhere to the coding standard too!